guides/javascript.md

```diff
@@ -0,0 +1,242 @@
+
+🚫
+
+``` javascript
+const removed = false;
+```
+
+✅
+
+``` javascript
+// Any of these would do
+const isRemoved = false;
+const wasRemoved = false;
+const hasBeenRemoved = false;
+const shouldBeRemoved = false;
+```
+
+If the name represents a negative statement, reword it into a positive statement and inverting the value assigned to the value.
+
+🚫
+
+``` javascript
+const notEnoughGas = false;
+```
+
+✅
+
+``` javascript
+const hasEnoughGas = true;
+```
+
+Take special note of variables which are created via React's `useState` hook.
+
+🚫
+
+``` javascript
+const [removed, setRemoved] = useState(false);
+```
+
+✅
+
+``` javascript
+const [isRemoved, setIsRemoved] = useState(false);
+```
+
+## 💡 Place names of secondary concepts first in boolean variable names
+
+When naming a boolean variable that includes a subject, the previous guideline suggests that you can place the verb at the beginning:
+
+``` javascript
+const isRecipientOwnedAccount = Boolean(ownedAccountName);
+```
+
+However, this naming strategy creates a point of friction for objects, arrays, or React components, where it may be desirable to sort identifiers alphabetically. In that case you could end up with something like:
+
+``` jsx
+<SenderToRecipient
+  isRecipientOwnedAccount={isRecipientOwnedAccount}
+  onClick={onClick}
+  recipientName={toName}
+  recipientNickname={toNickname}
+  senderAddress={fromAddress}
+  senderName={fromName}
+/>
+```
+
+It is potentially more "greppable" if properties that concern the same concept are kept together instead of separate. To address this, you may wish to place the subject of the variable name at the beginning:
+
+``` javascript
+const recipientIsOwnedAccount = Boolean(ownedAccountName);
+```
+
+This would result in:
+
+``` jsx
+<SenderToRecipient
+  recipientIsOwnedAccount={isRecipientOwnedAccount}
+  recipientName={toName}
+  recipientNickname={toNickname}
+  onClick={onClick}
+  senderAddress={fromAddress}
+  senderName={fromName}
+/>
+```
+
+## Use `async`/`await` syntax over `.then`/`.catch`
+
+Asynchronous code written using `async`/`await` syntax looks less complex and more straightforward than code written using `.then`/`.catch`. Additionally, using `async`/`await` leads to [better stack traces in Node and Chromium](https://mathiasbynens.be/notes/async-stack-traces), both of which use the V8 JavaScript engine, because when an asynchronous operation is `await`ed, the engine will remember the function where the `await` occurred, which means that it can place that function on the stack trace (otherwise, using `.then`/`.catch`, it would get lost).
+
+🚫
+
+``` javascript
+function makeRequest() {
+  return fetch("https://google.com")
```

Comment:
Nit: We use single quotes everywhere.
```suggestion
  return fetch('https://google.com')
```

Reply:
Good point! I need to add Prettier to this repo with the same settings that we usually use. I can fix this for now though.